server/src/storage/datastore.ts

```diff
@@ -123,6 +123,32 @@ const updateData = (original: StringKeyObject, updateRule: UpdateRule) => {
   original[updateRule.property] = value;
 };
 
+/**
+ * Updates an entity with the specified kind and id in a transaction
+ * according to the update rules.
+ * @param transaction Transaction that will carry out the update
+ * @param kind Kind of the entity to be updated
+ * @param id id of the entity to be updated
+ * @param updateRules Rules to update the entity
+ */
+const updateEntityInTransaction = async (
```

Comment:
Good job! This is a great way to refactor operations inside a transaction. Was also thinking of doing this, but hadn't figured out how to do it.